build_runner/lib/src/entrypoint/run.dart

```diff
@@ -18,19 +20,59 @@ import 'runner.dart';
 /// implies success.
 Future<int> run(List<String> args, List<BuilderApplication> builders) async {
   var runner = BuildCommandRunner(builders);
-  try {
-    var result = await runner.run(args);
-    return result ?? 0;
-  } on UsageException catch (e) {
-    print(ansi.red.wrap(e.message));
-    print('');
-    print(e.usage);
-    return ExitCode.usage.code;
-  } on ArgumentError catch (e) {
-    print(ansi.red.wrap(e.toString()));
-    return ExitCode.usage.code;
-  } on CannotBuildException {
-    // A message should have already been logged.
-    return ExitCode.config.code;
+  var resultCompleter = Completer<int>();
+
+  void safeComplete(int exitCode) {
+    if (resultCompleter.isCompleted) return;
+    resultCompleter.complete(exitCode);
+  }
+
+  unawaited(runZoned(() async {
```

Comment:
looks like the surrounding method doesn't need to be `async` and so wouldn't need the `unawaited`.